internal/keypool/deleted_key_cleanup_test.go

```
@@ -75,49 +110,159 @@ func TestHandleFailure_DeletedKeyDoesNotResurrectStoreHash(t *testing.T) {
+			length, err := memStore.LLen(activeKeysListKey)
+			require.NoError(t, err)
+			require.Zero(t, length, "deleted key ID must be removed from the active list")
+		})
+	}
+}
+
+func TestHandleFailure_DeletionDuringStoreWriteDoesNotRecreateHash(t *testing.T) {
 	provider, db, memStore := setupTestProvider(t)
-	defer provider.Stop()
+	provider.Stop()
+	defer memStore.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `memStore.Close` is not checked

(errcheck)